test-runner/src/main/scala/com/typesafe/spark/test/mesos/RolesSpec.scala

```diff
@@ -36,54 +35,13 @@ trait RoleSpecHelper {
 
     // pre-conditions
     if (cfg.role != "*") {
-      assert(m.slaves.flatMap { x => x.roleResources.map { y => y.roleName } }.contains(cfg.role), "Spark role should exist.")
-    }
-
-    val expectedUsedCpus = {
-      val m = MesosCluster.loadStates(mesosConsoleUrl)
-      val tmp = m.slaves.map { x => x.resources.cpu }.sum
-      if (isInClusterMode) {
-          tmp - 1 // minus the cpus owned by the Spark Cluster, which is started before the tests in cluster mode
-      }
-      else tmp
+      assert(m.slaves.flatMap { x => x.reservedResources.keys }.contains(cfg.role), "Spark role should exist.")
     }
 
-    val mesosUrl = mesosConsoleUrl
-    val accum = sc.accumulator[Double](0L, "cpuCounter")
-    val partitions = 40 // give it enough tasks
-
-    // use enough tasks to utilize all expected resources for tested role and *, pick a task in the middle and observe resources
-    val rdd = sc.makeRDD(1 to 100, partitions).mapPartitionsWithIndex{ (idx, iterator) =>
-
-      var counter = 1
-      val retries = 1000
-      val partitionNumber = partitions / 2
-      val timeToSleep = 10 // small resolution to catch the event of a fully utilized cluster
-      var max = 0
-
-      if (idx == partitionNumber) {
-        while (counter <= retries) {
-          val currentNumOfCpus = MesosCluster.loadStates(mesosUrl).sparkFramework.get.resources.cpu
-
-          if (max <= currentNumOfCpus){ max = currentNumOfCpus.toInt }
-
-          if (currentNumOfCpus != expectedUsedCpus) {
-            Thread.sleep(timeToSleep)
-            counter += 1
-          } else {
-            counter = retries + 1 // exit
-          }
-        }
-
-        if (idx == partitionNumber) {
-          accum += max
-        }
-      }
-      iterator
-    }
+    assert(m.sparkFramework.get.role == cfg.role)
 
+    val rdd = sc.makeRDD(1 to 5)
     val res = rdd.sum()
-    assert(accum.value == expectedUsedCpus, "All cpu resources should be utilized.")
```

Comment:
This doesn't test anything related to roles, and regardless, this isn't a valid assertion in an arbitrary cluster.  You can't say that Spark will use all the CPUs in the cluster when there are other frameworks running.